File: generate/reference/duality-of-compilation/ocaml/fibonacci_recursive.ml
let rec fibonacci i =
  if i = 0 then
    i
  else if i = 1 then
    i
  else
    fibonacci (i - 1) + fibonacci (i - 2)

let main n =
  fibonacci n

let () =
  match Sys.argv with
  | [| _ |] -> print_endline (string_of_int (main 10))
  | [| _; s |] -> (
      match int_of_string_opt s with
      | Some n -> print_endline (string_of_int (main n))
      | None -> print_endline "argument is not a number")
  | _ -> print_endline "arguments are too many"

File: generate/reference/duality-of-compilation/rust/fibonacci_recursive.rs
use std::env;

fn fibonacci(i: isize) -> isize {
    if i == 0 {
        i
    } else if i == 1 {
        i
    } else {
        fibonacci(i - 1) + fibonacci(i - 2)
    }
}

fn run(n: isize) -> isize {
    fibonacci(n)
}

fn main() {
    let args: Vec<String> = env::args().collect();

    let arg = match args.get(1) {
        None => 10,
        Some(val) => match val.parse::<isize>() {
            Ok(val) => val,
            Err(_) => {
                println!("argument is not a number");
                return;
            }
        },
    };

    println!("{}", run(arg));
}
